onnxruntime/contrib_ops/cpu/bert/attention.cc

```diff
@@ -115,151 +115,175 @@ Status Attention<T>::Compute(OpKernelContext* context) const {
   TensorShape output_shape(dims);
   Tensor* output = context->Output(0, output_shape);
 
-  const size_t element_size = sizeof(T);
-
-  // Use GEMM for fully connection.
-  int m = batch_size * sequence_length;
-  int n = 3 * hidden_size;
-  int k = hidden_size;
+  constexpr size_t element_size = sizeof(T);
 
   AllocatorPtr allocator;
   ORT_RETURN_IF_ERROR(context->GetTempSpaceAllocator(&allocator));
 
-  concurrency::ThreadPool* tp = context->GetOperatorThreadPool();
-
-  // STEP.1: gemm_data(BS, 3NH) = input(BS, NH) x weights(NH, 3NH) + bias(NH)
+  // STEP.1: gemm_data(BS, 3NH) = input(BS, NH) x weights(NH, 3NH) + bias(3NH)
   auto gemm_data = allocator->Alloc(batch_size * sequence_length * 3 * hidden_size * element_size);
   BufferUniquePtr gemm_buffer(gemm_data, BufferDeleter(allocator));
+  auto Q = reinterpret_cast<T*>(gemm_data);
+  auto K = Q + batch_size * sequence_length * hidden_size;
```

Comment:
store batch_size * sequence_length * hidden_size to a variable. It is used in multiple places in this function.